src/App.tsx

```diff
@@ -15,48 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortField = 'alph' | 'length' | '';
+
+const SORT_FIELD_ALPH = 'alph';
+const SORT_FIELD_LENTGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortField,
+  reverse: boolean,
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPH:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+
+      case SORT_FIELD_LENTGTH:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default:
+        preparedGoods = [...preparedGoods];
+    }
+  }
+
+  if (reverse) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField>('');
+  const [reverseField, setReverseField] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/96409563/23e707a3-5527-43da-8693-2d7948972b1a)